clang/test/CIR/IR/data-member-ptr.cir

```diff
@@ -0,0 +1,46 @@
+// RUN: cir-opt %s | cir-opt | FileCheck %s
+
+!s32i = !cir.int<s, 32>
+!ty_22Foo22 = !cir.struct<struct "Foo" {!s32i}>
+
+#global_ptr = #cir.data_member<0> : !cir.data_member<!s32i in !ty_22Foo22>
```

Comment:
Hey, I've stumbled upon this test when refactoring the `cir.const` assembly.
Can you explain what you were trying to test here? The reason I'm asking is that what you have here is an attribute alias, not a global value. That's the reason that the output MLIR is "inlining" it into:

```
// CHECK-NEXT:   cir.func @get_global_member(%arg0: !cir.ptr<!ty_22Foo22>) {
// CHECK-NEXT:     %0 = cir.const(#cir.data_member<0> : !cir.data_member<!s32i in !ty_22Foo22>) : !cir.data_member<!s32i in !ty_22Foo22>
// CHECK-NEXT:     %1 = cir.get_runtime_member %arg0[%0 : !cir.data_member<!s32i in !ty_22Foo22>] : !cir.ptr<!ty_22Foo22> -> !cir.ptr<!s32i>
// CHECK-NEXT:     cir.return
```

Reply:
@Lancern ^

Reply:
@orbiri Well this should just be an effort to save some typings when writing this test. But clearly it does little help, so you can change it if it's convenient for you.

Reply:
> @orbiri Well this should just be an effort to save some typings when writing this test. But clearly it does little help, so you can change it if it's convenient for you.

From the test name it seems like you tried to test the incorporation of some global property here, but if not - then `get_global_member` is 1:1 equivalent to  `get_runtime_member`, and I'd suggest to remove it.

If indeed you intended to test some global property, I'd suggest to revisit this test to make sure that it won't break over time 🙏

Reply:
> From the test name it seems like you tried to test the incorporation of some global property here, but if not - then get_global_member is 1:1 equivalent to get_runtime_member, and I'd suggest to remove it.

I checked this again and I'm sure they're 1:1 equivalent as you said. I'll remove it in an NFC PR if you request so.

Reply:
> I'll remove it in an NFC PR if you request so.

See #582 .